### ISSUE TITLE
no attribute 'output_layers_node_indices'

### ISSUE THREAD

**Issue by ycxia** [BenWhetton/keras-surgeon#8]
when i run lenet_mnist.py, something wrong:
Traceback (most recent call last):
  File "E:/keras-surgeon-master/keras-surgeon-master/src/kerassurgeon/examples/lenet_mnist.py", line 109, in <module>
    main()
  File "E:/keras-surgeon-master/keras-surgeon-master/src/kerassurgeon/examples/lenet_mnist.py", line 81, in main
    model = delete_channels(model, layer, channel_idx)
  File "D:\Anaconda3\lib\site-packages\kerassurgeon\operations.py", line 105, in delete_channels
    return surgeon.operate()
  File "D:\Anaconda3\lib\site-packages\kerassurgeon\surgeon.py", line 163, in operate
    enumerate(self.model.output_layers_node_indices)]
AttributeError: 'Sequential' object has no attribute 'output_layers_node_indices'
What should i do? thank you!

**Comment by adibmehrabi** [comment]
I have the same issue using Keras 2.2.0 and tensorflow 1.6.0...

**Comment by cvanea** [comment]
After having looked through the Keras source code, the file which contained the variable 'output_layers_node_indices' was removed on April 24th. I haven't tested it myself, but I think this would be an issue all the way until Keras version 2.1.6, which still has the variable. You could try reverting to 2.1.6 but it seems a fair bit has changed internally for 2.2.0 so depending on your code that may not be viable.

**Comment by cvanea** [comment]
it's a bit of a hack, but you can get around this by subclassing Surgeon and then changing the 'output_nodes' definition in 'operate' based on your known number of outputs.

For example, this works for binary output:
```python
output_nodes = [get_inbound_nodes(model._output_layers[0])[0]]
```
Hope that helps!

**Comment by adibmehrabi** [comment]
Thanks for your suggestions - I have been hacking it by setting the number of nodes as you mention :)

**Comment by BenWhetton** [comment]
Yeah, this is the second time I've been bitten by this kind of thing.
When I first made this library, I assumed that every non-protected attribute was part of the public API and would be subject to semantic versioning. Unfortunately, that doesn't seem to be true. I intentionally avoided using the tensors' `_keras_history` attribute because it is protected even though it would have made the code quite a bit easier to write.
That attribute seems to be sticking around for now so I've fixed this issue in [0.1.1](https://github.com/BenWhetton/keras-surgeon/releases/tag/0.1.1) using it.
